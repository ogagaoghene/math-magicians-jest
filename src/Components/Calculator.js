import React, { useState } from 'react';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [object, setObject] = useState({ total: 0, next: null, operation: null });
  const { total, next, operation } = object;

  const handleClick = (event) => {
    setObject((state) => calculate(state, event.target.value));
  };

  return (
    <div className="container">
      <form name="calculator">
        <div className="inputArea">
          <input type="text" name="userInput" placeholder="0" value={(total || '') + (operation || '') + (next || '')} disabled />
        </div>
        <div className="buttons">
          <div className="btnAC btnr1">
            <input type="button" value="AC" onClick={handleClick} />
          </div>
          <div className="btnSign btnr1">
            <input type="button" value="+/-" onClick={handleClick} />
          </div>
          <div className="btnMod btnr1">
            <input type="button" value="%" onClick={handleClick} />
          </div>
          <div className="btnDiv btnr1">
            <input type="button" value="&#247;" onClick={handleClick} />
          </div>
          <div className="btn7 btnr2">
            <input type="button" value="7" onClick={handleClick} />
          </div>
          <div className="btn8 btnr2">
            <input type="button" value="8" onClick={handleClick} />
          </div>
          <div className="btn9 btnr2">
            <input type="button" value="9" onClick={handleClick} />
          </div>
          <div className="btnX btnr1">
            <input type="button" value="x" onClick={handleClick} />
          </div>
          <div className="btn4 btnr2">
            <input type="button" value="4" onClick={handleClick} />
          </div>
          <div className="btn5 btnr2">
            <input type="button" value="5" onClick={handleClick} />
          </div>
          <div className="btn6 btnr2">
            <input type="button" value="6" onClick={handleClick} />
          </div>
          <div className="btnM btnr1">
            <input type="button" value="-" onClick={handleClick} />
          </div>
          <div className="btnAC btnr2">
            <input type="button" value="1" onClick={handleClick} />
          </div>
          <div className="btn2 btnr2">
            <input type="button" value="2" onClick={handleClick} />
          </div>
          <div className="btn3 btnr2">
            <input type="button" value="3" onClick={handleClick} />
          </div>
          <div className="btnA btnr1">
            <input type="button" value="+" onClick={handleClick} />
          </div>
          <div className="btn0 colspan">
            <input type="button" value="0" onClick={handleClick} />
          </div>
          <div className="btnDecimal btnr2">
            <input type="button" value="." onClick={handleClick} />
          </div>
          <div className="btnE btnr1">
            <input type="button" value="=" onClick={handleClick} />
          </div>
        </div>
      </form>
    </div>
  );
};

export default Calculator;
