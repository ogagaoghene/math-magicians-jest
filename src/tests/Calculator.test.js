import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../Components/Calculator';

describe('Snapshot testing', () => {
  it('Check if Calculator component is rendered', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
