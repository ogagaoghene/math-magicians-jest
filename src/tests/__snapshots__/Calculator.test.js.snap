// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing Check if Calculator component is rendered 1`] = `
<div
  className="container"
>
  <form
    name="calculator"
  >
    <div
      className="inputArea"
    >
      <input
        disabled={true}
        name="userInput"
        placeholder="0"
        type="text"
        value=""
      />
    </div>
    <div
      className="buttons"
    >
      <div
        className="btnAC btnr1"
      >
        <input
          onClick={[Function]}
          type="button"
          value="AC"
        />
      </div>
      <div
        className="btnSign btnr1"
      >
        <input
          onClick={[Function]}
          type="button"
          value="+/-"
        />
      </div>
      <div
        className="btnMod btnr1"
      >
        <input
          onClick={[Function]}
          type="button"
          value="%"
        />
      </div>
      <div
        className="btnDiv btnr1"
      >
        <input
          onClick={[Function]}
          type="button"
          value="รท"
        />
      </div>
      <div
        className="btn7 btnr2"
      >
        <input
          onClick={[Function]}
          type="button"
          value="7"
        />
      </div>
      <div
        className="btn8 btnr2"
      >
        <input
          onClick={[Function]}
          type="button"
          value="8"
        />
      </div>
      <div
        className="btn9 btnr2"
      >
        <input
          onClick={[Function]}
          type="button"
          value="9"
        />
      </div>
      <div
        className="btnX btnr1"
      >
        <input
          onClick={[Function]}
          type="button"
          value="x"
        />
      </div>
      <div
        className="btn4 btnr2"
      >
        <input
          onClick={[Function]}
          type="button"
          value="4"
        />
      </div>
      <div
        className="btn5 btnr2"
      >
        <input
          onClick={[Function]}
          type="button"
          value="5"
        />
      </div>
      <div
        className="btn6 btnr2"
      >
        <input
          onClick={[Function]}
          type="button"
          value="6"
        />
      </div>
      <div
        className="btnM btnr1"
      >
        <input
          onClick={[Function]}
          type="button"
          value="-"
        />
      </div>
      <div
        className="btnAC btnr2"
      >
        <input
          onClick={[Function]}
          type="button"
          value="1"
        />
      </div>
      <div
        className="btn2 btnr2"
      >
        <input
          onClick={[Function]}
          type="button"
          value="2"
        />
      </div>
      <div
        className="btn3 btnr2"
      >
        <input
          onClick={[Function]}
          type="button"
          value="3"
        />
      </div>
      <div
        className="btnA btnr1"
      >
        <input
          onClick={[Function]}
          type="button"
          value="+"
        />
      </div>
      <div
        className="btn0 colspan"
      >
        <input
          onClick={[Function]}
          type="button"
          value="0"
        />
      </div>
      <div
        className="btnDecimal btnr2"
      >
        <input
          onClick={[Function]}
          type="button"
          value="."
        />
      </div>
      <div
        className="btnE btnr1"
      >
        <input
          onClick={[Function]}
          type="button"
          value="="
        />
      </div>
    </div>
  </form>
</div>
`;
